yt_dlp/extractor/laracasts.py

```diff
@@ -0,0 +1,117 @@
+            **traverse_obj(episode, {
+                'id': ('id', {int}, {str_or_none}),
+                'webpage_url': ('path', {lambda x: urljoin('https://laracasts.com', x)}),
+                'title': ('title', {clean_html}),
+                'season_number': ('chapter', {int}),
+                'episode_number': ('position', {int}),
+                'description': ('body', {clean_html}),
+                'thumbnail': ('largeThumbnail', {url_or_none}),
+                'duration': ('length', {int_or_none}),
+                'date': ('dateSegments', 'published', {unified_strdate}),
+            }))
+
+
+class LaracastsIE(LaracastsBaseIE):
+    IE_NAME = 'laracasts'
+    _VALID_URL = r'https?://(?:www\.)?laracasts\.com/series/(?P<id>[\w\d-]+/episodes/[0-9]+)/?(?:[?#]|$)'
+    _TESTS = [{
+        'url': 'https://laracasts.com/series/30-days-to-learn-laravel-11/episodes/1',
+        'md5': 'c8f5e7b02ad0e438ef9280a08c8493dc',
+        'info_dict': {
+            'id': '922040563',
+            'title': 'Hello, Laravel',
+            'ext': 'mp4',
+            'duration': 519,
+            'date': '20240312',
+            'thumbnail': 'https://laracasts.s3.amazonaws.com/videos/thumbnails/youtube/30-days-to-learn-laravel-11-1.png',
+            'description': 'md5:ddd658bb241975871d236555657e1dd1',
+            'season_number': 1,
+            'season': 'Season 1',
+            'episode_number': 1,
+            'episode': 'Episode 1',
+            'uploader': 'Laracasts',
+            'uploader_id': 'user20182673',
+            'uploader_url': 'https://vimeo.com/user20182673',
+        },
+        'expected_warnings': ['Failed to parse XML'],  # TODO: Remove when vimeo extractor is fixed
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        return self.parse_episode(self.get_prop_data(url, display_id)['lesson'])
+
+
+class LaracastsPlaylistIE(LaracastsBaseIE):
+    IE_NAME = 'laracasts:series'
+    _VALID_URL = r'https?://(?:www\.)?laracasts\.com/series/(?P<id>[\w\d-]+)/?(?:[?#]|$)'
+    _TESTS = [{
+        'url': 'https://laracasts.com/series/30-days-to-learn-laravel-11',
+        'info_dict': {
+            'title': '30 Days to Learn Laravel',
+            'id': '210',
+            'thumbnail': 'https://laracasts.s3.amazonaws.com/series/thumbnails/social-cards/30-days-to-learn-laravel-11.png?v=2',
+            'duration': 30600.0,
+            'modified_date': '20240511',
+            'description': 'md5:27c260a1668a450984e8f901579912dd',
+            'categories': ['Frameworks'],
+            'tags': ['Laravel'],
+            'display_id': '30-days-to-learn-laravel-11',
+        },
+        'playlist_count': 30,
+    }]
+
+    def _entries(self, series, episode_count):
+        for current_episode in range(1, episode_count + 1):
+            webpage_url = f'https://laracasts.com/series/{series}/episodes/{current_episode}'
+            yield self.url_result(webpage_url, LaracastsIE)
```

Comment:
this method is never used?

Reply:
Ah, missed removing that during refactoring